shares/shares.go

```diff
@@ -14,70 +15,51 @@ type Share struct {
 	data []byte
 }
 
-func (s *Share) Namespace() (namespace.Namespace, error) {
-	if len(s.data) < namespace.NamespaceSize {
-		panic(fmt.Sprintf("share %s is too short to contain a namespace", s))
-	}
-	return namespace.From(s.data[:namespace.NamespaceSize])
-}
-
-func (s *Share) InfoByte() (InfoByte, error) {
-	if len(s.data) < namespace.NamespaceSize+ShareInfoBytes {
-		return 0, fmt.Errorf("share %s is too short to contain an info byte", s)
-	}
-	// the info byte is the first byte after the namespace
-	unparsed := s.data[namespace.NamespaceSize]
-	return ParseInfoByte(unparsed)
-}
-
 func NewShare(data []byte) (*Share, error) {
 	if err := validateSize(data); err != nil {
 		return nil, err
 	}
+	// TODO: we should also validate namespace
 	return &Share{data}, nil
 }
 
-func (s *Share) Validate() error {
-	return validateSize(s.data)
-}
-
 func validateSize(data []byte) error {
 	if len(data) != ShareSize {
 		return fmt.Errorf("share data must be %d bytes, got %d", ShareSize, len(data))
 	}
 	return nil
 }
 
-func (s *Share) Len() int {
-	return len(s.data)
+// Namespace returns the shares namespace
+// TODO: we could validate it the namespace in the constructor
```

Comment:
```suggestion
// TODO: we could validate the namespace in the constructor
```

Also similar, do we need an issue for this?

Reply:
It's the same issue just mentioned in two different places. I will create the issue. Thanks for reminding